_posts/2021-06-08-whats-coming-in-eslint-8.0.0.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title: What's coming in ESLint v8.0.0
+tags:
+  - major
+  - breaking changes
+---
+# What's coming in ESLint v8.0.0
+
+Last week, the Technical Steering Committee (TSC) met to finalize the features for ESLint v8.0.0. This feature set had not yet been locked down even as v8.0.0 work began and so this represented an important milestone that allows us to focus on getting this next major release shipped. You can see everything that is planned for v8.0.0 on our [project board](https://github.com/eslint/eslint/projects/8).
+
+We want to share with you the significant changes that will be coming in v8.0.0 so you can better prepare for the changes and understand why the changes are necessary.
+
+## Significant changes in v8.0.0
+
+The following changes are important to understand as we move towards v8.0.0.
+
+### Dropping support for Node.js v10.x and 13.x
+
+Node.js v10.x end-of-life was April 30, 2021 and Node.js v13.x end-of-life was June 1, 2021. As a result, ESLint v8.0.0 [removes support](https://github.com/eslint/eslint/issues/14023) for both of these Node.js versions.
+
+### Removing the `codeframe` and `table` formatters
+
+In an ongoing effort to reduce the install size of ESLint, we have decided to [remove the `codeframe` and `table` formatters](https://github.com/eslint/eslint/issues/14277) from the core of ESLint. Both of these formatters include dependencies that aren't used anywhere else in ESLint. If you are using these formatters currently, you'll need to install the standalone packages ([`eslint-formatter-codeframe`](https://www.npmjs.com/package/eslint-formatter-codeframe) and [`eslint-formatter-table`](https://www.npmjs.com/package/eslint-formatter-table)) for use with ESLint v8.0.0.
+
+### Support for ES2022 class fields and top-level `await`
+
+Support for ES2022 [comes to ESLint v8.0.0](https://github.com/eslint/eslint/issues/14343) with native support for class fields and top-level `await`.
+
+### Rules with suggestions now require the `meta.hasSuggestions` property
+
+Beginning in ESLint v8.0.0, rules that provide suggestions [must include a `meta.hasSuggestions: true` property](https://github.com/eslint/eslint/issues/14312). This allows ESLint to understand what a rule's capabilities are just by inspecting the rule object. Rules that provide suggestions without `meta.hasSuggestions: true` will cause an error.
+
+### Fixable rules now require `meta.fixable` property
+
+In ESLint v7.0.0 and earlier, it was possible to write a rule as a function instead of an object and to provide fixes without specifying a `meta.fixable` property. Beginning in v8.0.0, ESLint requires all rules that provide fixes to be specified in object format and [include `meta.fixable: true`](https://github.com/eslint/eslint/issues/13349).
```

Comment:
Not `true`, per the [specification](https://eslint.org/docs/developer-guide/working-with-rules#rule-basics) it should be string value `"code"` or `"whitespace"`.

Though, we're not validating the exact value - the only validation is that it must be truthy.

Reply:
I keep forgetting that. :) I’ll clarify.